File: DWG.cs
Console.Title = "DWG";

Console.ForegroundColor = ConsoleColor.Red;
Console.WriteLine(@"
             ______        ______ 
            |  _ \ \      / / ___|
            | | | \ \ /\ / / |  _ 
            | |_| |\ V  V /| |_| |
            |____/  \_/\_/  \____|
~===============~ DWG v1.0.3 ~===============~

");
Console.ResetColor();

Console.ForegroundColor = ConsoleColor.Cyan;
Console.WriteLine("Enter the amount of webhooks you want to generate:");
Console.ResetColor();

int webhook_count = 0;
while (true) {
	bool success = true;
	try {
		Console.Write("~> ");
		webhook_count = int.Parse(Console.ReadLine());
		if (webhook_count <= 0) {
			throw new Exception();
		}
	} catch (FormatException) {
		success = false;
		Console.ForegroundColor = ConsoleColor.Red;
		Console.WriteLine("Invalid amount of webhooks! Please try again.");
		Console.ResetColor();
	} catch (ArgumentNullException) {
		success = false;
		Console.ForegroundColor = ConsoleColor.Red;
		Console.WriteLine("Please enter a valid amount of webhooks! Please try again.");
		Console.ResetColor();
	} catch (Exception) {
		success = false;
		Console.ForegroundColor = ConsoleColor.Red;
		Console.WriteLine("Please enter a valid number of webhooks count. Please try again.");
		Console.ResetColor();
	}
	if (success) {
		break;
	}
}

static string RandomToken() {
	const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_";
	return new string(Enumerable.Repeat(chars, new Random().Next(67, 69)).Select(s => s[new Random().Next(s.Length)]).ToArray());
}

static string RandomNumber() {
	const string chars = "0123456789";
	return new string(Enumerable.Repeat(chars, 18).Select(s => s[new Random().Next(s.Length)]).ToArray());
}

if (File.Exists("webhooks.txt")) {
	Console.ForegroundColor = ConsoleColor.Yellow;
	Console.WriteLine("Cleaning up old webhooks...");
	Console.ResetColor();
	File.WriteAllText("webhooks.txt", "");
}
for (int i = 0; i <= webhook_count; i++) {
	Console.SetCursorPosition(0, Console.CursorTop);
	Console.Write($"Generating webhooks... | {i}/{webhook_count}");
	File.AppendAllText("webhooks.txt", $"\nhttps://discord.com/api/webhooks/{RandomNumber()}/{RandomToken()}");
	Thread.Sleep(10);
}

Console.ForegroundColor = ConsoleColor.Green;
Console.WriteLine($"\nSuccessfully generated {webhook_count} webhooks!");
Console.ResetColor();

Console.WriteLine("Press any key to terminate the program...");
Console.Write("...");
Console.ReadKey();

File: DWG.py
from os import system, path, getlogin
from colorama import Fore, init
from random import choice, randint

system("title Discord Webhooks Generator")
init(autoreset=True)

ver = 'v1.0.7 stable'
print(Fore.RED + f"\r\n             ______        ______ \r\n            |  _ \\ \\      / / ___|\r\n            | | | \\ \\ /\\ / / |  _ \r\n            | |_| |\\ V  V /| |_| |\r\n            |____/  \\_/\\_/  \\____|\r\n~===============~ DWG {ver} ~===============~\r\n\r\n")

print(Fore.CYAN + "Enter the amount of webhooks you want to generate:")
webhook_count = 0
while True:
    passed = True
    try:
        webhook_count = int(input("~> "))
    except ValueError:
        passed = False
        print(Fore.LIGHTRED_EX + "Please enter a valid number of webhooks!")
    else:
        if webhook_count == 0 or webhook_count <= 0:
            passed = False
            print(Fore.LIGHTRED_EX + "Please enter a valid number of webhooks!")

    if passed:
        break

high_performance = False
if webhook_count >= 100000:
    print(Fore.CYAN + "\nYou're trying to generate big amount of webhooks! Do you want to activate high performance mode? [y/n]")
    if input("~> ").lower() == "y":
        high_performance = True

def random_token():
    chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_"
    return ''.join(choice(chars) for _ in range(randint(67, 68)))

def random_number():
    chars = "0123456789"
    return ''.join(choice(chars) for _ in range(18))

if path.exists(f"C:/Users/{getlogin()}/Downloads/webhooks.txt"):
    print(Fore.LIGHTYELLOW_EX + "\nCleaning up old webhooks...")
    open(f"C:/Users/{getlogin()}/Downloads/webhooks.txt", "w").write("")
    print(Fore.LIGHTGREEN_EX + "Successfully cleaned up old webhooks!")

if high_performance:
    with open(f"C:/Users/{getlogin()}/Downloads/webhooks.txt", "a") as f:
        for i in range(webhook_count):
            f.write(f"\nhttps://discord.com/api/webhooks/{random_number()}/{random_token()}")
            print(f"Generating webhooks... | {round(i / webhook_count * 100)}%", end="\r")
else:
    with open(f"C:/Users/{getlogin()}/Downloads/webhooks.txt", "a") as f:
        for i in range(webhook_count):
            f.write(f"\nhttps://discord.com/api/webhooks/{random_number()}/{random_token()}")
            print(f"Generating webhooks... | {round(i / webhook_count * 100)}% ({i}/{webhook_count})", end="\r")

print(Fore.LIGHTGREEN_EX + f"Successfully generated {webhook_count} webhooks in your downloads folder!")
print(Fore.CYAN + "\nDo you want to open the webhooks.txt file? [y/n]")
if input("~> ").lower() == "y":
    system(f"start C:/Users/{getlogin()}/Downloads/webhooks.txt")
input(Fore.LIGHTBLACK_EX + "\nPress any key to terminate the program..." + Fore.RESET)
